README.md

```diff
@@ -1488,13 +1488,33 @@ Valid `opt` keys include:
     - `'number'`: synonymous for `number: true`, see [`number()`](#number)
     - `'string'`: synonymous for `string: true`, see [`string()`](#string)
 
-.parse(args)
+.parse(args, [fn])
 ------------
 
 Parse `args` instead of `process.argv`. Returns the `argv` object.
-
 `args` may either be a pre-processed argv array, or a raw argument string.
 
+A callback `fn` can optionally be provided as the second argument to `.parse()`.
+If a callback is given, it will be invoked with three arguments:
+
+1. `err`: populated if any validation errors raised while parsing.
+2. `argv`: the parsed argv object.
+3. `output`: any text that would have been output to the terminal, had a
+  callback not been provided.
+
+```js
+// providing the `fn` argument to `parse()` runs yargs in headless mode, this
+// makes it easy to use yargs in contexts other than the CLI, e.g., writing
+// a chat-bot.
+yargs()
```

Comment:
@scriptdaemon I love this idea, and think it would make things a lot more powerful ... I think I have an idea as to how to add this functionality without too major of an API change. What do you think of this:

```
yargs
  .command('foo')
  .parse('foo --bar', {user: 'bcoe}, function (err, argv, output) {
   
  })
```

where `{user: "bcoe"}` would be treated like passing in a configuration file.